src/web/app/pages-instructor/instructor-courses-page/instructor-courses-page.component.ts

```diff
@@ -292,10 +292,10 @@ export class InstructorCoursesPageComponent implements OnInit {
   /**
    * Permanently deletes a soft-deleted course in Recycle Bin.
    */
-  onDeletePermanently(courseId: string): void {
+  onDeletePermanently(courseId: string): Promise<void> {
```

Comment:
What's the goal of returning Promise here?

Reply:
The promise helps us to know if the function has finished deleting the course (because the ngbmodal result itself is a promise) -- otherwise, it would be hard for the tests to know when the async operations are done before testing for the desired changes. There doesn't seem to be any other way to do it that's as straightforward as using promises, so I went with this approach